Allow to be used in modules

hons82 opened the pull request:
As it uses GData in a dependency and that one uses libxml you're not able to use it within a static library or a framework

hons82 committed:
Allow to be used in modules

maxep commented:
Thanks @hons82

hons82 commented:
Actually I've seen that it's not the right way of doing that. So it's good that you didn't put it onto Cocoapods yet.

I've sent a [PullRequest](https://github.com/graetzer/GDataXML-HTML/pull/13) to the author of GDataXML-HTML which would really solve it. However there are still some issues with the imports of AFNetworking that I'll sent you as a PullRequest when the other library has been updated.

There is a similar issue in the imports in the files that are generated to be used with PicoKit which I'll try to fix too.

Hopefully you'll like my change ideas :-)
~Hons

maxep commented:
Hey @hons82 ,

Thanks very much for your contribution. Should I revert you pull request then ?
Will you try to use AF 2.+ ?

Regards,

Maxime

maxep commented:
FYI, I've managed to upgrade to AF 2.5.4, check latest code.

hons82 commented:
Wow. Nice
I'll check it.

My pull request can be reverted, actually I already did that in my fork, but didn't sent it as Pull Request. The problem needs to be solved in [GDataXML-HTML](https://github.com/graetzer/GDataXML-HTML/pull/13), however it doesn't seem to be accepted. Unfortunately there is no way to specify a dependency to a git repo in a podspec file, so currently I've to do this in my Podfile

```
pod 'GDataXML-HTML', :git => 'https://github.com/hons82/GDataXML-HTML'
pod 'PicoKit', :git => 'https://github.com/hons82/PicoKit'
```

to get it run. 

Currently I take my fork of your repo, because I'm using it in a Swift project with `use_frameworks!` so all the AfNetworking imports need to be like this `#import <AfNetworking/...>` instead of `#import "AfNetworking/..."` ([link to my changes](https://github.com/maxep/PicoKit/compare/master...hons82:master))

maxep commented:
I have changed imports to `<AFNetworking/AFNetworking.h>`, can you try it ?

I have also checked your pull request to GDataXML-HTML, to get it accepted I think you should reduce addition/deletion as much as you can, I think your modifications can be resumed to [this](https://github.com/maxep/GDataXML-HTML/commit/84c55051fe624523575ba7344129fc6266d295b9). I did it on my [fork](https://github.com/maxep/GDataXML-HTML), you can also try it if you want!

Thanks again,

Maxime

hons82 commented:
True… i removed some comments from the podspec to simplify it. 
Acctually if Apple would provide this module map a lot of pods would work without changes. 
Tomorrow in the office I'll try it

~hons

Sent from my iPhone

> On 15.07.2015, at 21:10, Maxime notifications@github.com wrote:
> 
> I have changed imports to <AFNetworking/AFNetworking.h>, can you try it ?
> 
> I have also checked your pull request to GDataXML-HTML, to get it accepted I think you should reduce addition/deletion as much as you can, I think your modifications can be resumed to this. I did it on my fork, you can also try it if you want!
> 
> Thanks again,
> 
> Maxime
> 
> —
> Reply to this email directly or view it on GitHub.

maxep commented:
Hi @hons82 

did you try this changes in your swift project ? with my GDataXML-HTML fork ?

Regards,

maxep commented:
By the mean time, I've release the v0.7.0, it includes the GDataXML-HTML files to get rid of the module problem.

hons82 commented:
Hi, sorry, that I didn't work on this for so long.
I'm about to change work place, so I had to setup a few things.

Now I just saw that the original library included my fix which finally should make it work. I just tried it in my swift project and it looks okay, so I send you  a PR #5
